### ISSUE TITLE
Enable non-sink functions within a DAG to write to KVS

### ISSUE THREAD

**Issue by cw75** [fluent-project/fluent#106]

**Comment by vsreekanti** [comment]
This is already possible. The only thing we have to change to enable this is to have the client library instrument reads/writes to track causal dependencies. Everything else already works.

**Comment by cw75** [comment]
The client should also be able to specify which function in the DAG want to write the result to KVS and what's the name of output key.

**Comment by vsreekanti** [comment]
How is that any different from calling `fluent.put(key, val)`?

**Comment by cw75** [comment]
I think letting users embed these calls into functions is fine. Maybe we should just do this: by default the result of a func is only passed to downstream funcs, but if users want it to also write to kvs, then they need to explicitly write `fluent.put(key, val)` inside the func.

**Comment by vsreekanti** [comment]
Yes, that's how it works right now. That's what I meant when I said the only thing that we needed to change was adding support for causal dependencies.
